sdk/core/core/CMakeLists.txt

```diff
@@ -72,4 +92,9 @@ endif()
 # make sure that users can consume the project as a library.
 add_library (az::core ALIAS ${TARGET_NAME})
 
+target_compile_definitions(${TARGET_NAME} PRIVATE "AZ_VERSION_MAJOR=${AZ_VERSION_MAJOR}")
+target_compile_definitions(${TARGET_NAME} PRIVATE "AZ_VERSION_MINOR=${AZ_VERSION_MINOR}")
+target_compile_definitions(${TARGET_NAME} PRIVATE "AZ_VERSION_PATCH=${AZ_VERSION_PATCH}")
+target_compile_definitions(${TARGET_NAME} PRIVATE "AZ_VERSION_PRERELEASE=\"${AZ_VERSION_PRERELEASE}\"")
```

Comment:
**Question:** Why does `AZ_VERSION_PRERELEASE` need quotes around it?

Reply:
It becomes a compiler flag which we want as a string.  This code has been removed as I'm no longer setting as compile flags